src/jdk.compiler/share/classes/com/sun/tools/javac/jvm/Gen.java

```diff
@@ -1285,6 +1285,11 @@ public void visitTry(JCTry tree) {
                     hasTry[0] = true;
                 }
 
+                @Override
+                public void visitSynchronized(JCSynchronized tree) {
+                    hasTry[0] = true;
+                }
+
```

Comment:
You can make this code slightly more efficient by using the fact that 'var' is enable to reference the type of an anonymous class.

```java
private boolean hasTry(JCSwitchExpression tree) {
            var treeScanner = new TreeScanner() {
                private boolean hasTry;                
 
                @Override
                public void visitTry(JCTry tree) {
                    hasTry = true;
                }

                @Override
                public void visitSynchronized(JCSynchronized tree) {
                    hasTry = true;
                }

                @Override
                public void visitClassDef(JCClassDecl tree) {
                }
                @Override
                public void visitLambda(JCLambda tree) {
                }
            };
            treeScanner.scan(tree);
            return treeScanner.hasTry;
        }
```

Reply:
nice suggestion